File: sol/solution/0001-0100/0013. Roman to Integer/Solution.kt
class Solution {
    fun romanToInt(s: String): Int {
        val valMap = mapOf('I' to 1, 'V' to 5, 'X' to 10, 'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000)
        var res = 0
        for (i in s.indices) {
            val curr = valMap[s[i]]!!
            val next = if (i + 1 < s.length) valMap[s[i + 1]]!! else 0
            res += if (curr < next) -curr else curr
        }
        return res
    }
}

File: sol/solution/0001-0100/0013. Roman to Integer/Solution.dart
class Solution {
  int romanToInt(String s) {
    final val = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000};
    int res = 0;
    for (int i = 0; i < s.length; i++) {
      int curr = val[s[i]]!;
      int next = i + 1 < s.length ? val[s[i + 1]]! : 0;
      res += (curr < next) ? -curr : curr;
    }
    return res;
  }
}